framework/src/org/checkerframework/framework/type/QualifierHierarchy.java

```diff
@@ -587,11 +601,18 @@ public AnnotationMirror findCorrespondingAnnotation(
     public AnnotationMirror getAnnotationInHierarchy(
             Collection<? extends AnnotationMirror> annos, AnnotationMirror top) {
         AnnotationMirror annoInHierarchy = null;
-        for (AnnotationMirror rhsAnno : annos) {
-            if (isSubtype(rhsAnno, top)) {
-                annoInHierarchy = rhsAnno;
+        AnnotationMirror polyAll = null;
+        for (AnnotationMirror anno : annos) {
+            boolean isSubtype = isSubtype(anno, top);
+            if (isSubtype && AnnotationUtils.areSameByClass(anno, PolyAll.class)) {
```

Comment:
I find this logic harder to parse than necessary. Would it be easier to have one test for `isSubtype` and then another for whether it is PolyAll?
The original code terminated the loop as soon as the first subtype was found. This always iterates over the whole list. Is this necessary?